odk/src/test/java/org/motechproject/odk/event/builder/impl/EventBuilderOdkTest.java

```diff
@@ -0,0 +1,139 @@
+package org.motechproject.odk.event.builder.impl;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.odk.constant.EventParameters;
+import org.motechproject.odk.domain.Configuration;
+import org.motechproject.odk.domain.FormDefinition;
+import org.motechproject.odk.domain.FormElement;
+import org.motechproject.odk.event.builder.EventBuilder;
+import org.motechproject.odk.event.builder.impl.EventBuilderOna;
+import org.motechproject.odk.parser.impl.XformParserODK;
+
+import java.io.File;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class EventBuilderOdkTest {
+
+    private Configuration configuration;
+    private String json;
+    private FormDefinition formDefinition;
+
+    @Before
+    public void setup() throws Exception{
+        configuration = new Configuration();
+        configuration.setName("configName");
+
+        File f = new File(getClass().getResource("/ona_nested_repeats.json").getFile());
+        json = FileUtils.readFileToString(f);
+
+        f = new File(getClass().getResource("/ona_nested_repeats.xml").getFile());
+        String xml = FileUtils.readFileToString(f);
+
+        formDefinition = new XformParserODK().parse(xml, configuration.getName());
+        alterFormDef(formDefinition);
+    }
+
+    @Test
+    public void testOnaNestedRepeats() throws Exception{
+
+        EventBuilder builder = new EventBuilderOna();
+        List<MotechEvent> eventList = builder.createEvents(json,formDefinition,configuration);
+        System.out.println();
+        assertEquals(eventList.size(), 7);
+        Iterator<MotechEvent> itr = eventList.iterator();
+
+        MotechEvent event = itr.next();
```

Comment:
spaces between parameters are missing in this file. We are not using checkstyle for test code, so this passes - it would be good to use spaces anyway.